torch/distributions/transforms.py

```diff
@@ -434,21 +434,24 @@ def __eq__(self, other):
         return isinstance(other, StickBreakingTransform)
 
     def _call(self, x):
-        shape = x.shape[:-1] + (1 + x.shape[-1],)
-        one = x.new([1]).expand(x.shape[:-1] + (1,))
-        numer = sigmoid(x)
-        denom = (1 - numer).cumprod(-1)
-        probs = torch.cat([numer, one], -1) * torch.cat([one, denom], -1)
-        return probs
+        offset = (x.shape[-1] + 1) - x.new([1]).expand(x.shape).cumsum(-1)
+        z = sigmoid(x - offset.log())
+        z_cumprod = (1 - z).cumprod(-1)
+        y = pad(z, (0, 1), value=1) * pad(z_cumprod, (1, 0), value=1)
+        return y
 
     def _inverse(self, y):
-        pmf = y
-        cmf = pmf.cumsum(-1)
-        sf = 1 - cmf
-        units = y[..., :-1] / sf[..., :-1]
-        return units.log()
+        shape = y.shape[:-1] + (y.shape[-1] - 1,)
+        offset = (shape[-1] + 1) - y.new([1]).expand(shape).cumsum(-1)
+        sf = (1 - y.cumsum(-1))[..., :-1]
+        x = y[..., :-1].log() - sf.log() + offset.log()
+        return x
 
-    # TODO implement .log_abs_det_jacobian()
+    def log_abs_det_jacobian(self, x, y):
```

Comment:
🎉 Thank you for implementing this!

How did you verify its correctness? We don't have great tests for multivariate `log_abs_det_jacobian`.

Reply:
My implementation is based on [Stan's reference manual](https://github.com/stan-dev/stan/releases/) (Section 35.6, pp. 408-410).

Yes, I noticed that `log_abs_det_jacobian` is tested only for univariate cases. Not sure if there is a way to have a generic set of tests for multivariate `log_abs_det_jacobian`. An indirect way would be to test `log_prob` of a distribution that uses this transform against an existing implementation. Unfortunately, `scipy.stats` doesn't have logistic-normal distribution.